nifi-nar-bundles/nifi-ccda-bundle/nifi-ccda-processors/src/main/java/org/apache/nifi/processors/ccda/ExtractCCDAAttributes.java

```diff
@@ -0,0 +1,383 @@
+        if ( flowFile == null ) {
+            return;
+        }
+
+        if(processMap.isEmpty()) {
+            getLogger().error("Process Mapping is not loaded");
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final Boolean skipValidation = context.getProperty(SKIP_VALIDATION).asBoolean();
+        final Boolean prettyPrinting = context.getProperty(PRETTY_PRINTING).asBoolean();
+
+        long start = System.currentTimeMillis(), total = start;
+
+        ClinicalDocument cd = loadDocument(session.read(flowFile), skipValidation); // Load and optionally validate CDA document
+
+        start = System.currentTimeMillis();
+        getLogger().info("Processing elements");
+        Map<String, Object> elements = processElement(null, cd); // Process CDA element using mapping data
+
+        start = System.currentTimeMillis();
+        getLogger().info("Generating JSON");
+
+        // Initialize GSON with optional pretty printing
+        Gson gson = prettyPrinting ? new GsonBuilder().setPrettyPrinting().create() : new GsonBuilder().create();
+        String json = gson.toJson(elements);
+        timingStats.add(String.format("JSON created in %d ms", System.currentTimeMillis() - start));
+        timingStats.add(String.format("CCDA Processed in %d ms", System.currentTimeMillis() - total));
+
+        logDetails(json);
+
+        flowFile = session.write(flowFile, new OutputStreamCallback() {
+
+            @Override
+            public void process(OutputStream output) throws IOException {
+                output.write(json.getBytes("UTF-8"));
+            }
+        });
+        flowFile = session.putAttribute(flowFile, CoreAttributes.MIME_TYPE.key(), APPLICATION_JSON);
+        flowFile = session.putAllAttributes(flowFile, attributes);
+
+        session.transfer(flowFile, REL_SUCCESS);
+
+    }
+
+    /**
+     * Process elements children based on the parser mapping.
+     * Any String values are added to attributes and JSON map
+     * For List, the processList method is called to iterate and process
+     * For an Object this method is called recursively
+     * While adding to the attributes the key is prefixed by parent
+     * @param parent    Parent key for this element, used as a prefix for attribute key
+     * @param element   Element to be processed
+     * @return          Map of processed data, value can contain String or Map of Strings
+     */
+    protected Map<String, Object> processElement(String parent, Object element) {
+        long start = System.currentTimeMillis();
+
+        Map<String, Object> map = new LinkedHashMap<String, Object>();
+        String name = element.getClass().getName();
+        Map<String, String> jexlMap = (HashMap<String, String>) processMap.get(name); // get JEXL mappings for this element
+        getLogger().debug("Processing " + name);
+
+        if (jexlMap == null) {
+            getLogger().info("Missing mapping for element " + name);

```

Comment:
Seems like this should be a warn() or error(). What's the impact here/what happens if this returns null? Seems like it'll mean addElement() never gets called so no attributes will be added; i.e., if there's a single missing mapping the processing short circuits but it still transfers to REL_SUCCESS. Is this right?